AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestExecutionStore.kt

```diff
@@ -0,0 +1,59 @@
+package dev.androidx.ci.testRunner
+
+import dev.androidx.ci.datastore.DatastoreApi
+import dev.androidx.ci.firebase.FirebaseTestLabApi
+import dev.androidx.ci.firebase.ToolsResultApi
+import dev.androidx.ci.gcloud.GoogleCloudApi
+
+internal class TestExecutionStore(
+    private val toolsResultApi: ToolsResultApi,
+    private val googleCloudApi: GoogleCloudApi,
+    firebaseProjectId: String,
+    datastoreApi: DatastoreApi,
+    firebaseTestLabApi: FirebaseTestLabApi,
+    gcsResultPath: String
+) {
+    private val testMatrixStore = TestMatrixStore(
+        firebaseProjectId = firebaseProjectId,
+        datastoreApi = datastoreApi,
+        firebaseTestLabApi = firebaseTestLabApi,
+        toolsResultApi = toolsResultApi,
+        resultsGcsPrefix = googleCloudApi.getGcsPath("aosp-ftl/$gcsResultPath")
+    )
+    private val testLabController = FirebaseTestLabController(
+        firebaseTestLabApi = firebaseTestLabApi,
+        firebaseProjectId = firebaseProjectId,
+        testMatrixStore = testMatrixStore
+    )
+
+    private suspend fun getTestExecutionStepsLogcats(
+        projectId: String,
+        historyId: String,
+        executionId: String
+    ): Map<String, String> {
+        val steps = toolsResultApi.listSteps(projectId, historyId, executionId).steps
+        val output = HashMap<String, String>()
```

Comment:
instead of filling into a map, you can use functional operators

something like:

```
return steps?.flatMap {
    step.testExecutionStep?.toolExecution?.toolOutputs ?: emptylist()
}.mapNotNull {
    if (it.testCase != null) { ... } else null
}
```

lmk if you need more help wrt kotlin parts.

Reply:
Changed it to return "step" here